x-pack/plugins/fleet/common/types/models/agent.ts

```diff
@@ -99,7 +111,40 @@ export interface CurrentUpgrade {
   startTime?: string;
 }
 
-// Generated from FleetServer schema.json
+interface FleetServerAgentComponentUnitInput {
+  id: string;
+  policy_id: string;
+  type: string;
+}
+
+interface FleetServerAgentComponentUnitOutput {
+  id: string;
+  policy_id: string;
+  type: string;
+  link: string;
+}
+
+interface FleetServerAgentComponentUnit {
+  id: string;
+  type: 'input' | 'output';
+  status: FleetServerAgentComponentStatus;
+  message: string;
+  input?: FleetServerAgentComponentUnitInput;
```

Comment:
Could we specify when the `input` or `output` will be present depending on the `type` field?

Reply:
I ended up removing `input/output` since the data we're actually getting doesn't include `input/output`. Looking at the [proto](https://github.com/elastic/elastic-agent-client/blob/b0328d2fe484f3ecfe6d52efdf18058431743e83/elastic-agent-client.proto#L296), it isn't defined there yet either 😅.